monai/apps/reconstruction/mri_dictionary.py

```diff
@@ -0,0 +1,285 @@
+    ) -> None:
+        MapTransform.__init__(self, keys, allow_missing_keys)
+        self.masker = EquispacedKspaceMask(
+            center_fractions=center_fractions, accelerations=accelerations, is_complex=is_complex
+        )
+
+    def set_random_state(
+        self, seed: Optional[int] = None, state: Optional[np.random.RandomState] = None
+    ) -> "EquispacedKspaceMaskd":
+        super().set_random_state(seed, state)
+        self.masker.set_random_state(seed, state)
+        return self
+
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, Tensor]:
+        """
+        Args:
+            data: is a dictionary containing (key,value) pairs from
+                the loaded dataset
+
+        Returns:
+            the new data dictionary
+        """
+        d = dict(data)
+        for key in self.key_iterator(d):  # key is typically just "kspace"
+            d[key + "_masked"], d[key + "_masked_ifft"] = self.masker(d[key])
+            d["mask"] = self.masker.mask  # type: ignore
+            meta = "_meta_dict"
+            d["target"] = d[key + meta][FastMRIKeys.RECON]  # type: ignore
+            d["filename"] = d[key + meta][FastMRIKeys.FILENAME]  # type: ignore
+        return d  # type: ignore
+
+
+class TargetBasedSpatialCropd(Cropd):
+    """
+    Dictionary-based wrapper of :py:class:`monai.transforms.SpatialCrop`.
+    This is similar to :py:class:`monai.transforms.SpatialCropd` but is
+    different since it crops the data based on the "Target" spatial size.
+    So for every new sample, the roi_center and roi_size will be different.
+    The first dimension is reserved for channel (e.g., could be num_slices).
+    The rest of the dimensions should be spatial. So the shape is (C,H,W,D)
+    for 3D samples.
+
+    Args:
+        keys: keys of the corresponding items to be transformed.
+            See also: :py:class:`monai.transforms.compose.MapTransform`
+        allow_missing_keys: don't raise exception if key is missing.
+    """
+
+    def __init__(self, keys: KeysCollection, allow_missing_keys: bool = False) -> None:
+
+        super().__init__(keys, cropper=None, allow_missing_keys=allow_missing_keys)  # type: ignore
+
+    def __call__(self, data: Mapping[Hashable, Tensor]) -> Dict[Hashable, Tensor]:
+        """
+        Args:
+            data: is a dictionary containing (key,value) pairs from
+                the loaded dataset
+
+        Returns:
+            the new data dictionary
+        """
+        d = dict(data)
+        for key in self.key_iterator(d):
+            # key is typically just "kspace_masked_ifft" for fastMRI data
+            image = d[key]
+            roi_size = d["target"].shape[1:]
+            roi_center = tuple(i // 2 for i in image.shape[1:])
+            cropper = SpatialCrop(roi_center=roi_center, roi_size=roi_size)
+            d[key] = convert_to_tensor(cropper(d[key]))
+        return d
+
+
+class DetailedNormalizeIntensityd(MapTransform):
```

Comment:
If I understand it correctly, I was wondering if it could be better to name it as `TargetBasedNormalizeIntensityd`

Reply:
Since here normalization is done based on input mean-std, it's not target-based in that sense. I'm thinking about `InputTargetNormalizeIntensity`, let me know if that doesn't work.

Reply:
I have two questions after reading the docstring. 1) Is it slice-wise NormalizeIntensity?, 2) Is  "target" a required key for input data dict? Thanks!